### ISSUE TITLE
Compass script links show a Lua error

### ISSUE THREAD

**Issue by pileofwolves** [pileofwolves/finmer#39]
When following a compass script link that does a `SetScene()` in order to complete a transition, the `GameplayScriptLibrary.ExportedSetScene` function will attempt to terminate the running script by yielding (and then never resuming the yielded coroutine, effectively stopping the script). However, Lua does not support yielding coroutines if there is a C frame in the call stack, since it has no way of capturing/restoring the state of that C stack frame when the coroutine is resumed (even though we don't do that in this case).

In effect, this causes this error message to be printed in the log:

> Script error in protected call: attempt to yield across metamethod/C-call boundary

The old engine (Update 1 and earlier) got around this by simply always throwing an error (with a special error message, which we would ignore / hide from the player). This does the same thing in that it immediately interrupts control flow and unwinds the call stack all the way back to the `lua_pcall` that started the script in the first place. I avoided this because it's a) inefficient as hell, and 2) very ugly to use exceptions for control flow.

In short, I need a different method to unwind the Lua call stack to return control flow to the initial `lua_pcall` invocation. Potential solutions:

- Investigate why Lua thinks there is an additional C stack frame, and remove it. Currently unclear why this is different from regular Turn calls.
- Reimplement using a coroutine, instead of doing a direct protected call, e.g. `lua_resume` instead of `lua_pcall`.
- Throw a script error like in Update 1. Ugly and non-performant, but will work as a last resort.

(Issue transcribed from Trello.)
